### DISCUSSION TITLE
difference between a static and a dynamic site both developed with html / css / js code etc ...

### DISCUSSION THREAD

**Discussion by Nemesi965** [community/community#31962] [Codespaces]
Hi guys, I'm a beginner and I have a doubt: what is the difference between a static and a dynamic site both developed with html / css / js code etc ...? Would you also have some examples to show me?

**Comment by MathiasReker** [comment]
A static site is client side only. A dynamic site is client side + backend.

**Reply by Nemesi965** [reply]
Ok, so is only this difference? Because some people talks about website with only image and text, other talks about website with database. In the front end it is not possible to see if a site is static or dynamic? It's so confusing!

**Reply by macknive** [reply]
@Nemesi965 You can feel on the frontend if it's static - it usually doesn't reload and just changing the elements on the frontend. Dynamic sites however use routes so it redirects you upon clicking a link making a page reload.

**Reply by airtower-luna** [reply]
> Dynamic sites however use routes so it redirects you upon clicking a link making a page reload.

That's inaccurate, dynamic sites can load data in the background, too.

The key difference is whether there's any code *running* on the server that generates pages or other responses (dynamic), or whether the server simply delivers *static* files (static).

**Reply by strong-728** [reply]
lets say about the websites, there are static and dynamic websites.
Static websites mean users can browse the wbesite without any changes and interactions so there are only html, css, js files.
But dynamic websites have both of frontend and backend so users can make the website's changes with click button or other events.
For this the website must have the backend and database shows the changed data for the clients.
hope this answer will be helpful for you.
Thanks.
